test_ui/product_id_flow/src/components/ProductSearchFlow.tsx

```diff
@@ -0,0 +1,333 @@
+import React, { useState, ChangeEvent } from "react";
+
+// =========== Interfaces ===========
+interface Product {
+  id: string;
+  product_team_id: string;
+  name: string;
+  ods_code: string;
+  status: string;
+  created_on: string;
+  updated_on: string | null;
+  deleted_on: string | null;
+  keys: any[];
+}
+
+interface ProductTeam {
+  product_team_id: string;
+  products: Product[];
+}
+
+interface SearchResult {
+  org_code: string;
+  product_teams: ProductTeam[];
+}
+
+interface ProductSearchResponse {
+  results: SearchResult[];
+}
+
+interface EnvironmentConfig {
+  environment: string;
+  apiKey: string;
+}
+
+// =========== Component ===========
+const ProductSearch: React.FC = () => {
+  const [currentStep, setCurrentStep] = useState(0);
+  const [environmentConfig, setEnvironmentConfig] = useState<EnvironmentConfig>(
+    {
+      environment: "",
+      apiKey: "",
+    }
+  );
+  const [organisationCode, setOrganisationCode] = useState("");
+  const [productTeamId, setProductTeamId] = useState("");
+  const [
+    searchResults,
+    setSearchResults,
+  ] = useState<ProductSearchResponse | null>(null);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+
+  const isEnvironmentConfigEnabled =
+    environmentConfig.environment.trim() !== "" &&
+    environmentConfig.apiKey.trim() !== "";
+
+  const handleSearch = async () => {
+    setLoading(true);
+    setError(null);
+
+    try {
+      // const queryParams = new URLSearchParams();
+      // if (organisationCode) queryParams.append("organisation_code", organisationCode);
+      // if (productTeamId) queryParams.append("product_team_id", productTeamId);
+
+      // const response = await fetch(
+      //   `https://${environmentConfig.environment}.api.service.nhs.uk/searchProduct?${queryParams.toString()}`,
+      //   {
+      //       method: "GET",
+      //       headers: {
+      //           Authorization: "letmein",
+      //           apikey: environmentConfig.apiKey,
+      //           version: "1",
+      //           "Content-Type": "application/json",
+      //       },
+      //   }
+      // );
+
+      // if (!response.ok) {
+      //   const errorBody = await response.text();
+      //   throw new Error(`HTTP error! status: ${response.status}, body: ${errorBody}`);
+      // }
+
+      // const responseData: ProductSearchResponse = await response.json();
+
+      const mockResponse: ProductSearchResponse = {
```

Comment:
presumably this will be removed?

Reply:
Yeah its dependant on Megans finished search code, that whole section can be switched out when shes finished - Theres no rush for this code to go through though to be fair so it could be wait to be complete?